app/Startup.cs

```diff
@@ -269,6 +271,24 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                     "{controller}/{action}/{caseId}/{locationId}/{containerId}/{bookingTime}"
                 );
             });
+
+            // Resolve the required services here
+            var memoryCache = app.ApplicationServices.GetRequiredService<IMemoryCache>();
```

Comment:
If you use IDistributedCache instead of IMemoryCache I think it should fix any load balancing issues by storing the data in a Postgres table instead of in memory.